src/I18nService.js

```diff
@@ -26,14 +26,33 @@ export class I18nService extends EventEmitter {
     }
   }
 
-  t(label, args) {
-    return this.translate(label, this.language, args)
+  t(label, args, mark) {
+    return this.translate(label, this.language, args, mark)
   }
 
-  translate(label, language, args) {
+  translate(label, language, args, mark) {
     this.ensureSupportedLanguage(language)
 
     const dictionary = this.getDictionary(language)
+    const marked = args?.[mark]
+
+    if (!isNaN(marked)) {
```

Comment:
Can't `mark` be undefined?

```suggestion

    if (mark && args?.[mark] && !isNaN(marked)) {
      const marked = args?.[mark]
```

Reply:
`mark` can be `undefined`, but the marked will be `undefined` too ->  `const marked = args?.[mark]`
> `if (mark && args?.[mark] && !isNaN(marked)) {`

this won't work because `mark` can be an index(for example 0)

Reply:
ok, then:
```
if (mark !== undefined && ...)
```

The idea is to fail fast if mark is not specified. The original code would test if we have some `args?.[mark]` even if we don't have the `mark` at all (i.e. for all our calls to `<T />` or `translate`). This can mean 100+ calls per page. It adds up...

Reply:
got it 
9036a18